### DISCUSSION TITLE
Does Detours provide a golang API?

### DISCUSSION THREAD

**Discussion by heimao63531** [microsoft/Detours#216] [Q&A] [answered]
Hi Detours team, 
i 'd love to use the Detours to hook winapi in goland. Does Detours support golang ?
for example , i want to add hook to the enable network adapter  api, how can i make it ?
v, err := wmi.CallMethod(nil, "Win32_NetworkAdapter.DeviceID="+fmt.Sprint(adapter.Index), "Enable", nil)

**Comment by sylveon** [comment, marked as answer]
Use [cgo](https://pkg.go.dev/cmd/cgo)
